src/basic/copy.c

```diff
@@ -386,14 +386,13 @@ static int fd_copy_directory(
                         continue;
                 }
 
-                if (buf.st_dev != original_device)
-                        continue;
-
                 if (S_ISREG(buf.st_mode))
                         q = fd_copy_regular(dirfd(d), de->d_name, &buf, fdt, de->d_name, override_uid, override_gid, copy_flags);
-                else if (S_ISDIR(buf.st_mode))
+                else if (S_ISDIR(buf.st_mode)) {
+                        if (buf.st_dev != original_device)
+                                continue;
```

Comment:
I'm sorry. Indeed, I was wrong about device nodes. I should have used `mknod` to check the behaviour of `stat`.

> I don't believe that we have any specific requirement to ignore bind mounted files.

Neither do I.